ext/mbstring/libmbfl/filters/mbfilter_ucs4.c

```diff
@@ -275,3 +295,10 @@ int mbfl_filt_conv_wchar_ucs4le(int c, mbfl_convert_filter *filter)
 
 	return c;
 }
+
+static int mbfl_filt_ident_ucs4(int c, mbfl_identify_filter *filter)
+{
+	/* Input should be a multiple of 4 bytes */
+	filter->status = (filter->status + 1) % 4;
+	return c;
```

Comment:
Hm, I'm not sure about this. This is not directly related to the identify filter, but more the input/output filters for UCS-4 in general. The input filter doesn't check the character range at all, and the output range checks 0..0x70000000. However, Unicode codepoints are limited to 0x110000, which is confusingly named MBFL_WCSPLANE_UTF32MAX. I think that the UCS4 input filter should be validating that range, and correspondingly the identify filter should do so as well.

Reply:
OK, done.